### ISSUE TITLE
BIOS Sledgehammer can't handle TPM update files with different BUILD version

### ISSUE THREAD

**Issue by texhex** [texhex/BiosSledgehammer#9]
The original TPM firmware files from HP (from SoftPaq 76423 - ftp://ftp.hp.com/pub/softpaq/sp76001-76500/sp76423.html) listed only one file for the upgrade from TPM 1.2/FW 6.41 to TPM 2.0:

```
TPM12_6.41.197.0_to_TPM20_7.41.2375.0.BIN
```

However, the TPM firmware files update from January 2017 (SoftPaq 78910 - ftp://ftp.hp.com/pub/softpaq/sp78501-79000/sp78910.html) now lists two different files for TPM 1.2/FW 6.41 to TPM 2.0:

```
TPM12_6.41.197.0_to_TPM20_7.61.2785.0.BIN
TPM12_6.41.198.0_to_TPM20_7.61.2785.0.BIN
```
Note that these two files are only different in the BUILD part of the version: 6.41.**197** vs. 6.41.**198**. It seems **.198** is only used when downgrading TPM 2.0/FW 7.61 to TPM 1.2 again as the archive contains on file for this: ``TPM20_7.61.2785.0_to_TPM12_6.41.198.0.BIN``.

The problem is that the [Win32_TPM](https://msdn.microsoft.com/en-us/library/windows/desktop/aa376484(v=vs.85).aspx) CIM class does not provide the BUILD part in the ``ManufacturerVersion`` field, only MAJOR.MINOR.

As we depend on this class, we can never detect if 6.41.**197** vs. 6.41.**198** is installed. This can lead to an error during the upgrade process.

**Comment by texhex** [comment]
I can confirm that machines, that are normally TPM 2.0 and are factory downgraded to TPM 1.2 by HP, will have 6.41.**198** installed. 

Seems we need to find a fix for this...

**Comment by texhex** [comment]
I expand the settings so a firmware entry can have ``.A`` or ``.B`` attached. Then we try to the update for A and if this does not work, for B.

**Comment by texhex** [comment]
First commit of the 2.46.x codebase today (5237d3dfad15014b0cc94d1eb5f2d60986c28cf0), hopefully this will enable the special handling for 6.41

**Comment by texhex** [comment]
Fixed in v2.46.1. Closing issue.
